src/main/java/org/broadinstitute/hellbender/engine/datasources/ReferenceMultiSource.java

```diff
@@ -20,14 +20,21 @@
 /**
  * Wrapper to load a reference sequence from the Google Genomics API, or a file stored on HDFS or locally.
  *
- * This class needs to be mocked, so it cannot be declared final.
+ * This class needs to subclassed by test code, so it cannot be declared final.
  */
 public class ReferenceMultiSource implements ReferenceSource, Serializable {
     private static final long serialVersionUID = 1L;
 
     private ReferenceSource referenceSource;
     private SerializableFunction<GATKRead, SimpleInterval> referenceWindowFunction;
 
+    // We can't use a real mock for this class in test code that requires an instance of this class,
+    // since in some cases they're used as a Spark broadcast variable. Mocks are mutated when they're
+    // accessed, which can result in ConcurrentModificationExceptions during serialization/broadcast.
+    // So we need to allow test code to create a subclass that provides a "fake" implementation of this
+    // class without actually being a mock.
```

Comment:
This comment might be more appropriate in a test class -- here you should just document concisely why the default constructor is protected.